app/models/page.rb

```diff
@@ -57,6 +57,24 @@ def meta_tags
     tag_names << plugin_names
   end
 
+  def dup
+    clone = super
+    clone.primary_image = nil
+    clone.slug = nil
+    clone
+  end
+
+  def number_of_pages_with_matching_title
+    Page.where(title: title).count
+  end
+
+  def slug_candidates
+    [
+      :title,
+      [:title, :number_of_pages_with_matching_title]
+    ]
+  end
```

Comment:
What do we think with this addition to `friendly_id`? A new page with an existing title will have a count appended to its slug: ''foo-bar-1"